sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -702,6 +702,18 @@ class DataFrameSuite extends QueryTest
       "The size of column names: 2 isn't equal to the size of metadata elements: 1"))
   }
 
+  test("SPARK-36642: withMetadata: replace metadata of a column") {
+    val metadata = new MetadataBuilder().putLong("key", 1L).build()
+    val df1 = sparkContext.parallelize(Array(1, 2, 3)).toDF("x")
+    val df2 = df1.withMetadata("x", metadata)
+    assert(df2.schema(0).metadata === metadata)
+
```

Comment:
nit, add a test which column existing metadata, test replace it with a new metadata

Reply:
I think that might not be necessary because by default (`df1`), column `"x"` has a metadata `Metadata.empty`, so the new metadata is already "replacing" the existing default metadata.